src/codegen/utilities/function_helpers.py

```diff
@@ -205,3 +258,26 @@ def instantiate_explicit_output_param(param):
         )
     elif param.ctypes_data_type == "ctypes.c_char_p":
         return f"{param.parameter_name} = ctypes.create_string_buffer(temp_size)"
+
+
+def get_return_value_for_func(func):
+    """Gets return value for the function."""
+    for output_parameter in func.base_parameters:
+        if output_parameter.direction == "out" and FUNCTION_RETURN_TYPE_MAP_SET.get(
+            output_parameter.type, None
+        ):
+            return FUNCTION_RETURN_TYPE_MAP_SET[output_parameter.type]
+    return None
+
+
+def get_input_params(func):
+    """Gets input parameters for the function."""
+    return (p for p in func.base_parameters if p.direction == "in")
+
+
+def get_skippable_param_for_func(func):
+    """Gets parameter name that needs to be skipped for the function."""
+    for param in func["parameters"]:
+        size = param.get("size", dict())
+        if size.get("mechanism", None) == "ivi-dance":
```

Comment:
Simplify: `if param.get("size", {}).get("mechanism") == "ivi-dance"`

Reply:
But `size` object is required within the if check. So, using it within the if check alone might make multiple get calls for `size` dict. 
However, I have optimized the code to some extent.

Reply:
Ok, that's fine.